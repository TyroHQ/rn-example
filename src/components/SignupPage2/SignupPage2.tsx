import React from "react";
import { Text } from "react-native";
import styled from "styled-components/native";
import { Placeholder } from "../Placeholder";

const SignupPage2Container = styled.View`
  background-color: lightgrey;
  padding: 10px;
`;

export class SignupPage2 extends React.Component<{}> {
  static defaultProps: SignupPage2["props"];
  render() {
    return (
      <Placeholder>
        page
        <Placeholder> Header </Placeholder>
        <Placeholder>
          Content
          <Placeholder>Account Verification</Placeholder>
          <Placeholder>
            For password and account security verification
          </Placeholder>
          <Placeholder orientation="row">
            <Placeholder>Country code</Placeholder>
            <Placeholder>Phone number</Placeholder>
          </Placeholder>
          <Placeholder>Enter button</Placeholder>
        </Placeholder>
      </Placeholder>
    );
  }
}
