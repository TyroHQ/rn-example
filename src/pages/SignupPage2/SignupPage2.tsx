import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";
import { Placeholder } from "@src/components/Placeholder";

const SignupPage2Container = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class SignupPage2 extends React.Component<{}> {
  static defaultProps: SignupPage2["props"];
  render() {
    return (
      <Placeholder>
        page
        <Placeholder> Header </Placeholder>
        <Placeholder>
          Content
          <Placeholder>Account Verification</Placeholder>
          <Placeholder>
            For password and account security verification
          </Placeholder>
          <Placeholder orientation="row">
            <Placeholder orientation="column">
              <Placeholder>Country code</Placeholder>
              <Placeholder>Input field</Placeholder>
            </Placeholder>
            <Placeholder orientation="column">
              <Placeholder>Phone number</Placeholder>
              <Placeholder>Input field</Placeholder>
            </Placeholder>
          </Placeholder>
          <Placeholder>Enter button</Placeholder>
        </Placeholder>
      </Placeholder>
    );
  }
}
