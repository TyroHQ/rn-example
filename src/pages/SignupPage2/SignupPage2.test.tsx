import { shallow } from "enzyme";
import * as usage from "./SignupPage2.usage";

describe("<SignupPage2 />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
