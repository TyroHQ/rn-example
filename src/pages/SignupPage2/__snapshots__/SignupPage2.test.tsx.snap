// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupPage2 /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  page
  <Placeholder
    orientation="row"
  >
     Header 
  </Placeholder>
  <Placeholder
    orientation="row"
  >
    Content
    <Placeholder
      orientation="row"
    >
      Account Verification
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      For password and account security verification
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      <Placeholder
        orientation="column"
      >
        <Placeholder
          orientation="row"
        >
          Country code
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          Input field
        </Placeholder>
      </Placeholder>
      <Placeholder
        orientation="column"
      >
        <Placeholder
          orientation="row"
        >
          Phone number
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          Input field
        </Placeholder>
      </Placeholder>
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Enter button
    </Placeholder>
  </Placeholder>
</Placeholder>
`;
